src/niswitch/system_tests/test_system_niswitch.py

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/python
+
+import niswitch
+import pytest
+import warnings
+
+
+@pytest.fixture(scope='function')
+def session():
+    with niswitch.Session('', False, True, 'Simulate=1, DriverSetup=topology:2737/2-Wire 4x64 Matrix') as simulated_session:
+        yield simulated_session
+
+
+# Basic Use Case Tests
+def test_relayclose(session):
+    relay_name = 'kr0c0'
+    assert session.get_relay_position(relay_name) == niswitch.RelayPosition.OPEN
+    session.relay_control(relay_name, niswitch.RelayAction.CLOSE_RELAY)
+    assert session.get_relay_position(relay_name) == niswitch.RelayPosition.CLOSED
+    assert relay_name
+
+
+def test_channel_connection(session):
+    channel1 = 'c0'
+    channel2 = 'r0'
+    assert session.can_connect(channel1, channel2) == niswitch.PathCapability.PATH_AVAILABLE
+    session.connect(channel1, channel2)
+    assert session.can_connect(channel1, channel2) == niswitch.PathCapability.PATH_EXISTS
+    session.disconnect(channel1, channel2)
+    assert session.can_connect(channel1, channel2) == niswitch.PathCapability.PATH_AVAILABLE
+    session.connect(channel1, channel2)
+    assert session.can_connect(channel1, channel2) == niswitch.PathCapability.PATH_EXISTS
+    session.disconnect_all()
+    assert session.can_connect(channel1, channel2) == niswitch.PathCapability.PATH_AVAILABLE
+
+
+# Attribute Tests
+def test_viboolean_attribute(session):
+    session.interchange_check = False
+    assert session.interchange_check is False
+    session.interchange_check = True
+    assert session.interchange_check is True
+
+
+def test_vistring_attribute(session):
+    assert 'NI PXIe-2737' == session.instrument_model
+
+
+def test_viint32_attribute(session):
+    assert session.channel_count > 0
```

Comment:
FWIW if you are simulating a specific model, you could very well do a stricter check.
(up to you)

Reply:
Fixed